rusk-recovery/src/state.rs

```diff
@@ -125,58 +126,121 @@ where
 
     info!("{} network state", theme.action("Storing"));
 
-    let root = hex::encode(network.root());
-
-    if root != EXPECTED_ROOT {
-        error!("{} expected state root failed", theme.action("Checking"));
-        error!("{} {}", theme.error("Expected"), EXPECTED_ROOT);
-        error!("{} {}", theme.error("Computed"), root);
-        return Err("state root mismatch".into());
-    }
-
-    info!("{} {}", theme.action("Root"), root);
+    info!("{} {}", theme.action("Root"), hex::encode(network.root()));
 
     let state_id = network.persist(ctor).expect("Error in persistence");
 
     Ok(state_id)
 }
 
-pub fn exec(overwrite: bool) -> Result<(), Box<dyn std::error::Error>> {
+pub fn exec(build: bool) -> Result<(), Box<dyn Error>> {
     let theme = Theme::default();
 
     info!("{} Network state", theme.action("Checking"));
     let state_path = rusk_profile::get_rusk_state_dir()?;
     let id_path = rusk_profile::get_rusk_state_id_path()?;
 
-    let has_state = state_path.exists() && id_path.exists();
+    // if the state already exists in the expected path stop.
+    if state_path.exists() && id_path.exists() {
```

Comment:
This is similar - if not the exact same - as the suggestion made by @vlopes11 when we were talking about this change. While this would indeed be useful, the same can be achieved by merely deleting the state (or just the ID), and calling `make state` to redownload or build accordingly.

I suggest we tackle this in a separate issue.

Reply:
Opened new [issue](#518).